src/app/feed/feed.component.ts

```diff
@@ -86,10 +96,14 @@ export class FeedComponent implements OnInit, AfterViewInit, OnDestroy {
 	private queryFromURL(): void {
 		this.__subscriptions__.push(
 			this.route.queryParams
-					.subscribe((params: Params) => {
-						const queryParam = params['query'] || '';
-						this.search(queryParam);
```

Comment:
No you can't remove that it is still getting called on query change from the header.